core/src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -2334,3 +2335,26 @@ strip eq arr@Array{..} = do
             then getLast prev low
             else return cur
         else return cur
+
+-- | Bubble sorts the elements.
+--
+-- /Pre-release/
+{-# INLINE bubbleAsc #-}
+bubbleAsc :: (MonadIO m, Storable a) =>
+    Int -> (a -> a -> Ordering) -> Array a -> m ()
+bubbleAsc idx cmp0 arr =
+    do
+    let l = length arr
+    go idx l
+
+    where
+    go i l =
+        when ((i + 1) < l) $ do
```

Comment:
Assuming grow by appends as I assumed in an earlier comment, we should keep going back to the first index.